zebra-consensus/src/block/check.rs

```diff
@@ -94,14 +91,27 @@ pub fn subsidy_is_valid(block: &Block, network: Network) -> Result<(), BlockErro
         )
     } else if halving_div.count_ones() != 1 {
         unreachable!("invalid halving divisor: the halving divisor must be a non-zero power of two")
-    } else if height < canopy_activation_height {
+    } else if subsidy::founders_reward::founders_reward_active(height, network) {
         // Founders rewards are paid up to Canopy activation, on both mainnet and testnet
         let founders_reward = subsidy::founders_reward::founders_reward(height, network)
             .expect("invalid Amount: founders reward should be valid");
         let matching_values = subsidy::general::find_output_with_amount(coinbase, founders_reward);
 
+        // TODO: no validation yet, just calling founders reward address functions
+        let founders_reward_address =
+            subsidy::founders_reward::founders_reward_address(height, network)
+                .expect("we should have an address");
+        let matching_address = subsidy::founders_reward::find_output_with_address(
+            coinbase,
+            founders_reward_address,
+            network,
+        );
+
         // TODO: the exact founders reward value must be sent as a single output to the correct address
-        if !matching_values.is_empty() {
+        if !matching_values.is_empty()
+            && !matching_address.is_empty()
+            && matching_values == matching_address
+        {
             Ok(())
         } else {
             Err(SubsidyError::FoundersRewardNotFound)?
```

Comment:
Should we split this error into:
* matching value not found
* matching address not found

Reply:
There is another option if we do this:

- value and address found but they are not in the same output.

Reply:
Sounds good!

Reply:
Added this at https://github.com/ZcashFoundation/zebra/pull/1170/commits/6b36a34a4be40f6ad344452935935b5674781f51
There are a few `Todo` comments in the testing for test each of the founders reward errors, will do that in a separated commit.

Reply:
Modified the `founders_reward_validation_failure()` test case to trigger the 3 founder reward errors at https://github.com/ZcashFoundation/zebra/pull/1170/commits/22407e6689dd7593301655d7839aef8582f266ed